### ISSUE TITLE
Change Request: `"rule_name": "disable"` combine with `reportUnusedDisableDirectives`

### ISSUE THREAD

**Issue by loynoir** [eslint/eslint#18341]
### ESLint version

v8.57.0

### What problem do you want to solve?

When  `eslint /path/to/reproduce.ts` with below config, eslint will not report error.

```js
    files: ['/path/to/reproduce.ts'],
    rules: {
      'rule_that_was_needed_to_turn_off_but_currently_dont_need_to_turn_off': 'off'
    },
    linterOptions: {
      reportUnusedDisableDirectives: 'error'
    }
```

### What do you think is the correct solution?

When  `eslint /path/to/reproduce.ts` with below config, eslint will report error about `rule_that_was_needed_to_turn_off_but_currently_dont_need_to_turn_off`.

```js
    files: ['/path/to/reproduce.ts'],
    rules: {
      'rule_that_was_needed_to_turn_off_but_currently_dont_need_to_turn_off': 'disable'
    },
    linterOptions: {
      reportUnusedDisableDirectives: 'error'
    }
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
Thanks for the suggestion. I'm having trouble understanding the problem you're trying to solve and also how what you're proposing would help. Please describe the problem in more detail.

**Comment by loynoir** [comment]
Given config

```js
    linterOptions: {
      reportUnusedDisableDirectives: 'error'
    }
```

Below code will report error,  when code does not violate `xxx1` to `xxx5`.

```js
/* eslint-disable xxx1 */
/* eslint-disable xxx2 */
/* eslint-disable xxx3 */
/* eslint-disable xxx4 */
/* eslint-disable xxx5 */

code
```

But this makes code ugly.

I prefer

```js
    rules: {
      'xxx1': 'off',
      'xxx2': 'off',
      'xxx3': 'off',
      'xxx4': 'off',
      'xxx5': 'off',
    },
    linterOptions: {
      reportUnusedDisableDirectives: 'error'
    }
```

But when code don't violate rule `xxx1` to `xxx5`, I expect eslint to report disable rules are unused.

So, I propose a new value `disable`.

Before proposal, available values are `'error' | 'off' | 'warn'`.

After proposal, available values are `'error' | 'off' | 'warn' | 'disable'`.

Value `'xxx1': 'disable'`

- is treated as same as `/* eslint-disable xxx1 */`, but let code looks less ugly.

- is better than `'xxx1': 'off'`, because it is reported when code does not violate rule `xxx1`.

**Comment by nzakas** [comment]
> **But the problem is**, when code don't violate rule `xxx1` to `xxx5`, I expect eslint to report disable rules are unused.

Just to make sure I understand: you're saying that you want a way to say that a rule (`xxx1`) that is set to `"off"` can be removed because it would not cause any errors?

**Comment by loynoir** [comment]
> Just to make sure I understand: you're saying that you want a way to say that a rule (`xxx1`) that is set to `"off"` can be removed because it would not cause any errors?

Yes.

But also consider the fact of the plugin ecosystem

```js
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error"
```

I propose a rule (`xxx1`) that is set to `"disable"` can be removed because it would not cause any errors.

**Comment by nzakas** [comment]
Okay, thanks for confirming.

I'm not sure what you're proposing makes sense. If the rule is set to "disable", we would actually still need to execute the rule to determine if it produced any errors, which incurs a performance cost. So it's not actually disabled at all. The only difference is that when the rule does not report errors, that itself would be reported as an unused rule.

The confusion between "disable" and "off" would also be a key concern. I understand you're proposing "disable" so that it would still be reported by "reportUnusedDisableDirectives", but I think overall this is confusing.

So, I don't think the proposal as stated is something we want to implement.

This seems targeted at an edge case, and it's pretty easy to tell if a rule would result in an error by temporarily turning it on.

At this point, we're not going to pursue implementing such a change.
